cmd/find.go

```diff
@@ -0,0 +1,454 @@
+/*
+ * Minio Client (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// All mc utilities are stored under the cmd directory
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+	"github.com/minio/minio/pkg/probe"
+	"github.com/minio/minio/pkg/wildcard"
+)
+
+// FindMSG holds JSON and string values for printing
+type FindMSG struct {
+	Path string `json:"path"`
+}
+
+// String calls tells the console what to print and how to print it
+func (f FindMSG) String() string {
+	return console.Colorize("Find", f.Path)
+}
+
+// JSON formats output to be JSON output
+func (f FindMSG) JSON() string {
+	f.Path = "path"
+	jsonMessageBytes, e := json.Marshal(f)
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// NameFlag pattern matches off of the base of the filepath
+func NameFlag(path, pattern string) (bool, error) {
```

Comment:
The function name should be more appropriate `NameFlag` means its a flag not a matcher function.  We should name them appropriately such as 

```
func nameMatch(...)
func pathMatch(...)
func regexMatch(...)
```